libdnf5/transaction/transaction_replay.cpp

```diff
@@ -0,0 +1,432 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "../base/transaction_impl.hpp"
+#include "utils/string.hpp"
+
+#include "libdnf5/comps/group/query.hpp"
+#include "libdnf5/rpm/package_query.hpp"
+
+#include <json.h>
+#include <libdnf5/comps/environment/query.hpp>
+#include <libdnf5/comps/group/package.hpp>
+#include <libdnf5/transaction/transaction_replay.hpp>
+
+#include <utility>
+
+namespace {
+
+void raise_or_warn(bool warn_only, const std::string & msg) {
+    if (warn_only) {
+        //TODO(amatej): logger
+        printf("problem: %s\n", msg.c_str());
+    } else {
+        //TODO(amatej): throw custom TransactionReplay exception
+        throw std::runtime_error(msg);
+    }
+}
+
+}  // namespace
+
+namespace libdnf5::transaction {
+
+TransactionReplay::TransactionReplay(
+    const libdnf5::BaseWeakPtr & base, std::filesystem::path filename, bool ignore_installed)
```

Comment:
I am not sure whether it is good to do all of the logic in a constructor. Additionally, it stops on the first issue and this is a difference to goal.resolve().

Reply:
When there is any problem with file (permissions, format, ...) it is file to throw an error.